test/MUXControls.Test/Common/GamepadHelper.cs

```diff
@@ -54,14 +54,41 @@ public class GamepadHelper
     {
         public static void PressButton(UIObject obj, GamepadButton button)
         {
-            Log.Comment("Pressing Gamepad button: {0}", button);
+            var keyInputReceivedUIObject = FindElement.ById("__KeyInputReceived");
+
+            if(keyInputReceivedUIObject != null)
+            {
+                var keyInputReceivedCheckbox = new CheckBox(keyInputReceivedUIObject);
+
+                if (keyInputReceivedCheckbox.ToggleState == ToggleState.On)
+                {
+                    keyInputReceivedCheckbox.Uncheck();
+                }
+                Verify.IsTrue(keyInputReceivedCheckbox.ToggleState == ToggleState.Off);
+            }
+
+            Log.Comment("Pressing Gamepad button: {0}", button);    
             KeyboardInput[] array = new KeyboardInput[1];
             array[0].virtualKeyCode = (ushort)button;
             InternalNativeMethods.InjectKeyboardInput(array, 1);
 
-            // Without a delay, the input gets dropped.
-            Wait.ForMilliseconds(50);
 
+            // Wait until app reports it received the input
+            if(keyInputReceivedUIObject == null)
+            {
+                // Fallback if we don't find out helper
+                Wait.ForMilliseconds(50);
+            }
+            else
+            {
+                var keyInputReceivedCheckbox = new CheckBox(keyInputReceivedUIObject);
+                while (keyInputReceivedCheckbox.ToggleState != ToggleState.On)
+                {
+                    Wait.ForMilliseconds(10);
+                }
+            }
```

Comment:
I prefer using something like this [https://github.com/microsoft/microsoft-ui-xaml/blob/8ce19103946d40f0fbf043d137802b5f36215666/dev/SwipeControl/SwipeControl_InteractionTests/SwipeControlTests.cs#L1916](https://github.com/microsoft/microsoft-ui-xaml/blob/8ce19103946d40f0fbf043d137802b5f36215666/dev/SwipeControl/SwipeControl_InteractionTests/SwipeControlTests.cs#L1916)

Reply:
Switched to that. Is there any documentation on this? I can't seem to find any helpful documents on the Microsoft.Windows.Apps.Test.Foundation namespace or TAEF 🤔

Reply:
I think unfortunately not, @kmahone might know something

Reply:
Also just to be sure, TryWait in this case tries to either wait for the value to change or continue after 50 milliseconds right?

Reply:
You should be able to see the source by using F12 in VS.  I think that after the timespan this will throw, and I think there is an optional parameter to make it continue without throwing instead

Reply:
Hitting F12 only gives me this:
![image](https://user-images.githubusercontent.com/16122379/90570583-4e289e80-e1b0-11ea-9bbc-92e52af1c0f2.png)